chain/client/src/chunk_executor_actor.rs

```diff
@@ -0,0 +1,533 @@
+                block.header(),
+                &chunk_headers,
+                prev_block.header(),
+                is_new_chunk,
+            )?;
+
+            // If we don't care about shard we wouldn't have relevant incoming receipts.
+            let is_me = true;
+            if !self
+                .shard_tracker
+                .cares_about_shard_this_or_next_epoch(me, block_hash, shard_id, is_me)
+            {
+                continue;
+            }
+            // TODO(spice-resharding): We may need to take resharding into account here.
+            let receipt_proofs = self.get_incoming_receipts(prev_hash, shard_id)?;
+            let incoming_receipts = Some(&*receipt_proofs);
+
+            let storage_context =
+                StorageContext { storage_data_source: StorageDataSource::Db, state_patch };
+
+            let cached_shard_update_key =
+                Chain::get_cached_shard_update_key(&block_context, chunk_headers, shard_id)?;
+
+            let job = self.get_update_shard_job(
+                me,
+                cached_shard_update_key,
+                block_context,
+                chunk_headers,
+                shard_index,
+                &prev_block,
+                ApplyChunksMode::IsCaughtUp,
+                incoming_receipts,
+                storage_context,
+            );
+            match job {
+                Ok(Some(job)) => jobs.push(job),
+                Ok(None) => {}
+                Err(e) => panic!("{e:?}"),
+            }
+        }
+
+        let apply_result =
+            do_apply_chunks(BlockToApply::Normal(*block.hash()), block.header().height(), jobs);
+        let apply_result = apply_result.into_iter().map(|res| (res.0, res.2)).collect_vec();
+        let results = apply_result.into_iter().map(|(shard_id, x)| {
+            if let Err(err) = &x {
+                tracing::warn!(target: "chunk_executor", ?shard_id, hash = %block.hash(), %err, "error in applying chunk for block");
+            }
+            x
+        }).collect::<Result<Vec<_>, Error>>()?;
+
+        for result in &results {
+            let (shard_uid, apply_result) = match result {
+                ShardUpdateResult::NewChunk(NewChunkResult {
+                    shard_uid,
+                    gas_limit: _,
+                    apply_result,
+                }) => (shard_uid, apply_result),
+                ShardUpdateResult::OldChunk(OldChunkResult { shard_uid, apply_result }) => {
+                    (shard_uid, apply_result)
+                }
+            };
+            // TODO(spice-resharding): Make sure this logic is correct with resharding.
+            let shard_id = shard_uid.shard_id();
+            let shard_index = shard_layout.get_shard_index(shard_id)?;
+            let chunk_header =
+                chunk_headers.get(shard_index).ok_or(Error::InvalidShardId(shard_id))?;
+
+            let receipt_proofs = make_outgoing_receipts_proofs(
```

Comment:
I don't think we need the chunk header anymore - just the epoch ID?

Reply:
Only shard_id and shard_layout is required so I changed this accordingly as part of [`ff7fe8a` (#13637)](https://github.com/near/nearcore/pull/13637/commits/ff7fe8a54d011c45b5bced06ae2b78377001efe7).